bin/vault_utils/kv_engine.py

```diff
@@ -24,5 +24,51 @@ def secret(self, path):
 
 class VaultKVSecret(VaultSecret):
 
+    def __init__(self, vault_engine, path, version=None):
+        super(VaultKVSecret, self).__init__(vault_engine, path)
+
+        # explicit version defined
+        self._version = version
+
+        # attempt to use a requested "version 0"
+        if self._version is None:
```

Comment:
I'd have to look at the Vault docs, to figure out if this should just be a simple falsy check to also include the case when self._version is (explicitly) 0... 

Or if we should blow up if version is specified and not >= 1...

Reply:
This code doesn't attempt to enforce which versions are valid, though you're right that it could.  I don't want to set it to a simple falsy check, because that would return the latest version number even when explicitly asking for an invalid secret version number.

I'm also not sure it's wrong to just let the request go to vault at the time of query, and let vault respond that the version number isn't valid.  That said I need to fix exception handling in general, but that's not in scope for pre-.conf release.

Later in `previous_version` I _do_ consider that versions only go back to `1`.

Reply:
I'm going to close this for now, as the scope is more than I want to bite off at this time.  If it's something you strongly think should be done, maybe we should open an issue to track it.